authentication/serializers.py

```diff
@@ -191,3 +192,27 @@ class Meta:
 
     def get_is_connected(self, obj):
         return bool(obj.access_token and obj.access_token_secret)
+
+
+class ENSConnectionSerializer(BaseThirdPartyConnectionSerializer):
+    class Meta:
+        model = ENSConnection
+        fields = "__all__"
+        read_only_fields = [
+            "created_on",
+            "pk",
+            "user_profile",
+            "title",
+        ]
+
+    def is_valid(self, raise_exception=False):
+        super_is_validated = super().is_valid(raise_exception)
+        user_profile = self.context.get("request").user.profile
+        address = self.validated_data.get("user_wallet_address")
+        is_address_valid = user_profile.owns_wallet(address)
+        if not is_address_valid and raise_exception:
+            raise serializers.ValidationError(
+                {"user_wallet_address": "user_wallet_address is not owned by user"}
+            )
+
+        return is_address_valid and super_is_validated
```

Comment:
I think this part of the code is duplicated more than three times. Please refactor functions that have the same logic to reduce code duplication.